Bottom: 58e9028a1bf54ebf5798d89fc1e3f82eb66eb09e
Top:    8c588cffdadc406224075f7eb95e482dcfdbae10
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-22 14:46:15 -0500

Fix mariadb bind ip address


---
diff --git a/roles/mariadb/tasks/main.yml b/roles/mariadb/tasks/main.yml
index 61b296f3..966c9a93 100644
--- a/roles/mariadb/tasks/main.yml
+++ b/roles/mariadb/tasks/main.yml
@@ -20,15 +20,14 @@
 - name: Open the firewall
   firewalld: service=mysql permanent=true state=enabled
   when: mariadb_open_firewall
-- name: Set host ssh ListenAddress
+- name: Set host mysqld bind-address
   template:
     src: mariadb-bind-address.j2
     dest: /etc/my.cnf.d/mariadb-bind-address.cnf
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
-
+  register: bind_cnf
 - name: Restart MariaDB service
   service:
     name: mariadb
     enabled: yes
     state: restarted
-  when: ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
+  when: bind_cnf.changed
diff --git a/roles/mariadb/templates/mariadb-bind-address.j2 b/roles/mariadb/templates/mariadb-bind-address.j2
index f30ea73a..831edde0 100644
--- a/roles/mariadb/templates/mariadb-bind-address.j2
+++ b/roles/mariadb/templates/mariadb-bind-address.j2
@@ -1,3 +1,3 @@
 # Bind the database only to the host port
 [mysqld]
-bind-address = {{ ansible_br0["ipv4"]["address"] }}
+bind-address = {{ ansible_default_ipv4["address"] }}
